### ISSUE TITLE
training an english to german language using attention

### ISSUE THREAD

**Issue by mpizosdim** [google/trax#1161]
### Description
I am trying locally to train an english to german language model with attention. 
Training and Validation loss seems logical. However whenever I try to predict next symbol I get the same result always: a dash symbol.

Can someone explain me the reasons that it is happening? What is the 'casual' way to debug a trax model (check results of multiplications, results of each layer, weights of each layer etc.)?

Code is taken from coursera's course of NLP specialization course 4 where I successfully passed the exercize with the same code and running on coursera's notebook environment it gives normal results.

Repository with my code: https://github.com/mpizosdim/trax_translation_with_attention
...
Requirements: https://github.com/mpizosdim/trax_translation_with_attention/blob/master/requirements.txt
```
OS: Ubuntu 18.04

$ pip freeze | grep trax
trax==1.3.6

$ pip freeze | grep tensor
mesh-tensorflow==0.1.17
tensorboard==2.3.0
tensorboard-plugin-wit==1.7.0
tensorflow==2.3.1
tensorflow-datasets==4.0.1
tensorflow-estimator==2.3.0
tensorflow-metadata==0.24.0
tensorflow-text==2.3.0

$ pip freeze | grep jax
jax==0.2.4
jaxlib==0.1.56

$ python -V
Python 3.6.9
